pkg/encoding/simple8b/encoding.go

```diff
@@ -520,8 +524,8 @@ func canPack(src []uint64, n, bits int) bool {
 
 	max := uint64((1 << uint64(bits)) - 1)
 
-	for i := 0; i < end; i++ {
-		if src[i] > max {
+	for _, s := range src[:n] {
```

Comment:
Are `n` and `end` the same value?

Reply:
Agree, the logic is strange, with regard to `n` and `end`.

Lets walk through a pseudo-proof.

On entry to `canPack`, the first precondition is 

```
if n ≥ len(src) then immediately exit
```

therefore, the condition `n < len(src)` must be `true` in order to reach:

```go
end := len(src)
```

Next test, by substituting `end` for `len(src)`: 

```go
if n < len(src) {
    end = n
}
```

Given `n < len(src)` is always true, `end` will always equal `n`.

Reply:
looks like `end` can be removed.